### PR TITLE
ISPN-10778 Cache Store Default Segmented

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-10778
https://issues.jboss.org/browse/ISPN-10780

### CODE REVIEW

**Comment by wburns** [comment]
I have a feeling some non core tests may fail as well, so preview for now.

**Comment by wburns** [comment]
Okay the tests should be fixed 🤞 The new thread leak ones are preexisting and should be handled in https://issues.jboss.org/browse/ISPN-10812

File: persistence/jpa/src/main/java/org/infinispan/persistence/jpa/configuration/JpaStoreConfigurationBuilder.java [hunk 1]
```diff
@@ -50,6 +54,10 @@ public JpaStoreConfigurationBuilder storeMetadata(boolean storeMetadata) {
 
    @Override
    public void validate() {
+      Boolean segmented = attributes.attribute(SEGMENTED).get();
+      if (segmented == null || segmented) {
+         throw CONFIG.storeDoesNotSupportBeingSegmented(JpaStore.class.getSimpleName());
+      }
```
Comment (ryanemerson) [inline persistence/jpa/src/main/java/org/infinispan/persistence/jpa/configuration/JpaStoreConfigurationBuilder.java]: Shouldn't we set SEGMENTED to false automatically in the builder of store's which don't support segmented? That way users don't have to explicitly set it to false.

File: core/src/test/java/org/infinispan/persistence/file/SingleFileStoreStressTest.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ protected void clearTempDir() {
    protected EmbeddedCacheManager createCacheManager() throws Exception {
       location = TestingUtil.tmpDirectory(SingleFileStoreStressTest.class);
       ConfigurationBuilder builder = new ConfigurationBuilder();
-      builder.persistence().addSingleFileStore().location(this.location).purgeOnStartup(true);
+      builder.persistence().addSingleFileStore().location(this.location).purgeOnStartup(true).segmented(false);
```
[the remark this answers is not in the document]
Comment (wburns) [reply core/src/test/java/org/infinispan/persistence/file/SingleFileStoreStressTest.java]: TBH I can't find why this was changed. I don't see a failure on CI.

File: persistence/jpa/src/main/java/org/infinispan/persistence/jpa/configuration/JpaStoreConfigurationBuilder.java (hunk 1 above)
Comment (wburns) [reply persistence/jpa/src/main/java/org/infinispan/persistence/jpa/configuration/JpaStoreConfigurationBuilder.java]: You actually suggested this behavior :) 

> Ryan Emerson: I'm in favour of being explicit and throwing the error
> Ryan Emerson: That might make the user more aware of the limitations of a non-segmented store

File: core/src/test/java/org/infinispan/persistence/support/AsyncStoreWithoutEvictionFunctionalTest.java [hunk 3]
```diff
@@ -33,14 +35,45 @@
...
 
    private DefaultCacheManager configureCacheManager(boolean async) throws Exception {
 
       GlobalConfiguration glob = new GlobalConfigurationBuilder().defaultCacheName("cache").nonClusteredDefault().build();
       SingleFileStoreConfigurationBuilder bld = new ConfigurationBuilder()
             .clustering().cacheMode(CacheMode.LOCAL)
             .transaction().transactionMode(TransactionMode.NON_TRANSACTIONAL)
-            .persistence().passivation(false).addSingleFileStore().preload(false).purgeOnStartup(false);
+            .persistence().passivation(false)
+            .addSingleFileStore().preload(false).purgeOnStartup(false).segmented(segmented).location(tmpDirectory);
```
[the remark this answers is not in the document]
Comment (wburns) [reply core/src/test/java/org/infinispan/persistence/support/AsyncStoreWithoutEvictionFunctionalTest.java]: I was just leaving the test how it was before.

File: persistence/jpa/src/main/java/org/infinispan/persistence/jpa/configuration/JpaStoreConfigurationBuilder.java (hunk 1 above)
Comment (ryanemerson) [reply persistence/jpa/src/main/java/org/infinispan/persistence/jpa/configuration/JpaStoreConfigurationBuilder.java]: O yeah :see_no_evil:  ... sorry about that. Lets stick with what you have then :slightly_smiling_face:

File: core/src/test/java/org/infinispan/persistence/file/SingleFileStoreStressTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (wburns) [reply core/src/test/java/org/infinispan/persistence/file/SingleFileStoreStressTest.java]: I don't want to change this though if it is the only issue remaining. We can change this back after FINAL.

**Comment by ryanemerson** [comment]
@wburns Are the `ClusteredConditionalCommandPassivationTest` failures related? They appear on the last couple of CI runs.

**Comment by wburns** [comment]
> @wburns Are the `ClusteredConditionalCommandPassivationTest` failures related? They appear on the last couple of CI runs.

They are technically intermittent. The problem is we need to tweak the queue size for the test setup. They happen on master as well, I think this PR just makes them happen more often.

**Comment by wburns** [comment]
I added just a small blurb to the upgrading guide, no other changes.

**Comment by ryanemerson** [comment]
Thanks @wburns

**Pull request merged**